ape_hardhat/providers.py

```diff
@@ -104,12 +119,16 @@ def connect(self):
                 # was already running.
                 logger.info(f"Connecting to existing Hardhat node at port '{self.port}'.")
         else:
-            for _ in range(self.config.process_attempts):
+            for _ in range(self.config.process_attempts):  # type: ignore
                 try:
                     self._start_process()
                     break
+                except HardhatNotInstalledError:
+                    # Is a sub-class of `HardhatSubprocessError` but we to still raise
+                    # so we don't keep retrying.
+                    raise
```

Comment:
Note: this is the same thing as not having caught the error. Meaning, it will raise the `HardhatNotInstalledError` that it caught **and** preserve the exception stack